mlir/lib/Conversion/GPUToNVVM/LowerGpuOpsToNVVMOps.cpp

```diff
@@ -376,17 +376,44 @@ struct LowerGpuOpsToNVVMOpsPass
     LLVMTypeConverter converter(m.getContext(), options);
     configureGpuToNVVMTypeConverter(converter);
     RewritePatternSet llvmPatterns(m.getContext());
+    LLVMConversionTarget target(getContext());
+
+    if (!filterDialects.empty()) {
+      for (StringRef dialectName : filterDialects) {
+        Dialect *dialect = getContext().getLoadedDialect(dialectName);
+        // Dialect may not be loaded if it wasn't used in source module, ignore.
```

Comment:
What do you think about always querying `getContext().getLoadedDialects()` and filtering from there? I think this would help merge these two code paths a bit

Reply:
done